eeprog.c

```diff
@@ -140,14 +141,15 @@ int read_from_eeprom(struct eeprom *e, int addr, int size, int hex)
 	return 0;
 }
 
-int write_to_eeprom(struct eeprom *e, int addr)
+int write_to_eeprom(struct eeprom *e, int addr, int timeout)
 {
 	int c;
 	while((c = getchar()) != EOF)
 	{
 		print_info(".");
 		fflush(stdout);
 		die_if(eeprom_write_byte(e, addr++, c), "write error");
+		die_if(eeprom_wait_ready(e, timeout), "write timeout");
```

Comment:
I would call eeprom_wait_ready only if -t is used so it'll be backwards compatible and it won't break with devices that don't support I2C_SMBUS_QUICK (just in case).

Thanks for the PR